src/base/PhpBlockView.php

```diff
@@ -254,9 +264,69 @@ public function extraValue($key, $defaultValue = null, $template = false)
      * ```
      *
      * @return \luya\web\View The global application View Object which is also the same as the layout or cmslayout.
+     * @deprecated deprecated since 1.0.5 and will be removed in 1.1.0
      */
     public function getAppView()
     {
-        return Yii::$app->getView();
+        trigger_error('The getAppView() has been deprecated and will be removed in version 1.1.0. Use the view object to register assets, js and files.', E_USER_DEPRECATED);
+    
+        return $this;
+    }
+
+    /**
+     * @return array
+     * @since 1.0.5
+     */
+    public function getBlockAssets()
+    {
+        return [
+            'metaTags' => $this->metaTags,
+            'linkTags' => $this->linkTags,
+            'cssFiles' => $this->cssFiles,
+            'css' => $this->css,
+            'jsFiles' => $this->jsFiles,
+            'js' => $this->js,
+        ];
+    }
+
+    /**
+     * @param array $blockAssets
+     * @param array $assetBundles
+     * @throws \yii\base\InvalidConfigException
+     * @since 1.0.5
+     */
+    public static function registerToAppView(array $blockAssets, array $assetBundles)
+    {
+        $appView = Yii::$app->view;
+
+        foreach ($blockAssets as $attribute => $blockAsset) {
+            if (!empty($blockAsset)) {
+                
+                if ($attribute == 'js' || $attribute == 'jsFiles') {
+                    /**
+                     * js and jsFiles must keep the array keys as position and have subarray
+                     * @see \yii\web\View::POS_HEAD
+                     */
+
+                    $appAssets = &$appView->{$attribute};
+                    
+                    foreach ($blockAsset as $key => $value) {
+                        if (isset($appAssets[$key])) {
+                            $appAssets[$key] = array_merge($appAssets[$key], $value);
+                        }
+                        else {
+                            $appAssets[$key] = $value;
+                        }
+                    }
+                }
+                else {
```

Comment:
we do write `} else {` on the same line, see psr2 => https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#11-example